### DISCUSSION TITLE
how to add the ads_plugin.py within the FSLeye (FSLVm7_64)

### DISCUSSION THREAD

**Discussion by binghe-c** [neuropoly/axondeepseg#562] [Bug reports, suggestions for code refactoring]
Recently, I have installed a virtual machine and setup the FSLeye within the virtual machine.
However, when I try to open the ads_plugin.py within the FSLeye (FSLVm7_64), it shows that An error occurred installing the plugin file. Details: No  module same 'AxonDeepSeg'. 
May I know what is the possible reasons for this issue?

https://axondeepseg.readthedocs.io/en/latest/documentation.html#installation

In FSLeyes, do the following: - Click on file -> load plugin - Select ads_plugin.py (found in AxonDeepSeg folder) - When asked Install permanently click on yes.
![image](https://user-images.githubusercontent.com/69069748/133011725-5a868abc-bdb5-468b-9129-ee568384f71c.png)

**Comment by mathieuboudreau** [comment]
Hi @binghe-c,

As I understand it, you already had an FSLeyes installation on your computer, and tried to load our plugin in it.

If that's correct, your issue is that you actually have to install a new FSLeyes instance inside a virtual environment that you've installed the complete AxonDeepSeg package. I would recommend to start [here](https://axondeepseg.readthedocs.io/en/latest/documentation.html#miniconda) and work yourself down the page.

I only noticed now that if you click directly on the [Graphical User Interface section of the documentation](https://axondeepseg.readthedocs.io/en/latest/documentation.html#graphical-user-interface-gui-optional) (as you might have), there is no warning that you have to have installed ADS already for it to work. I'll open an issue for it.

**Reply by binghe-c** [reply]
Hi Mathiew, 
Actually I have install the ubuntu and FSL inside the virtual machine separately. Do I need to add another ADS inside the Ubuntu as well?
![image](https://user-images.githubusercontent.com/69069748/133183131-d94a4c5f-d2db-4e69-851e-af39cf8aa911.png)

**Reply by Stoyan-I-A** [reply]
What you have here is actually two Virtual Machines (one called `Ubuntu 64-bits` and the other called `FSLVm7_64`). But yes, you do need to install AxonDeepSeg in Ubuntu 64-bits if this is the Virutal Machine that you want to use.

**Comment by binghe-c** [comment]
I try to install the ADS inside the miniconda (within the virtual machine), but it show the following:
![image](https://user-images.githubusercontent.com/69069748/133183967-832930b5-2577-48ae-be9a-606a012b3d3c.png)
![image](https://user-images.githubusercontent.com/69069748/133184065-cbec7907-03ff-4381-bdb4-09fa3dbdb967.png)

**Reply by Stoyan-I-A** [reply]
~~I wonder if this error happens because you already tried to install AxonDeepSeg inside the `ads_venv` environment. I am not sure. If that is the case, this could be a bug on our side. I need to test this.~~

In the meantime, you can try to install AxonDeepSeg inside of a new Conda environment (you can call it `ads_venv2` for example):

```
cd axondeepseg
conda env create -f environment.yml -n ads_venv2
conda activate ads_venv2
pip install -e .
```

**Reply by Stoyan-I-A** [reply]
I just tested the installation on Ubuntu and I did not encounter this issue.

~~Can you check if your virtual machine has a sufficient amount of disk space that can be used? Normally, AxonDeepSeg should use less than 5 GB (including all the packages that are required). However, virtual machines will require more than that because the files used for the operating system are take up more space.~~

~~I think that allocating 30 GB to a virtual machine is a good amount.~~

**Reply by Stoyan-I-A** [reply]
Looking back at the error message, @hermancollin noticed that there seems to be a problem with the `requests` module.

You can try the instructions I wrote earlier and see if it helps:

```
cd axondeepseg
conda env create -f environment.yml -n ads_venv2
conda activate ads_venv2
pip install -e .
```

**Reply by binghe-c** [reply]
I try the updated the instructions and shows the following errors:
![image](https://user-images.githubusercontent.com/69069748/133404895-bf13d528-c336-49c7-8222-493dcf8b23c7.png)
![image](https://user-images.githubusercontent.com/69069748/133405089-55b74b73-57da-4b6a-9a3d-f6c142503814.png)

**Reply by hermancollin** [reply]
@binghe-c from your screenshot it seems you tried our suggestion inside Windows. Could you please try inside your Linux VM?

To be clear, normally, you wouldn't need to manually install FSL yourself because it is included in the ADS installation. These would be the steps to follow:
1. Create your virtual machine. As @Stoyan-I-A mentioned earlier, make sure to give it enough storage space
2. As indicated in the installation manual, run the following commands inside your VM:
install miniconda:
```bash
cd ~
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh
bash ~/miniconda.sh -b -p $HOME/miniconda
echo ". ~/miniconda/etc/profile.d/conda.sh" >> ~/.bashrc
source ~/.bashrc
```
fetch the axondeepseg source files
```bash
git clone https://github.com/neuropoly/axondeepseg.git
cd axondeepseg
```
create virtual environment which will contain python version and all dependencies:
```bash
conda env create -f environment.yml
```
activate virtual environment and install ADS (with FSLeyes included)
```bash
conda activate ads_venv
pip install -e .
```
As you might imagine, the order of all these commands is important.
